packages/flutter_tools/test/general.shard/build_system/targets/localizations_test.dart

```diff
@@ -0,0 +1,118 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/build_system/depfile.dart';
+import 'package:flutter_tools/src/build_system/targets/localizations.dart';
+
+import '../../../src/common.dart';
+import '../../../src/context.dart';
+
+void main() {
+  testWithoutContext('generateLocalizations forwards arguments correctly', () async {
+    final FileSystem fileSystem = MemoryFileSystem.test();
+    final Logger logger = BufferLogger.test();
+    final FakeProcessManager processManager = FakeProcessManager.list(<FakeCommand>[
+      const FakeCommand(
+        command: <String>[
+          'dart',
+          'dev/tools/localization/bin/gen_l10n.dart',
+          '--arb-dir=arb',
+          '--template-arb-file=example.arb',
+          '--output-localization-file=bar',
+          '--untranslated-messages-file=untranslated',
+          '--output-class=Foo',
+          '--header-file=header',
+          '--header=HEADER',
+          '--use-deferred-loading',
+          '--preferred-supported-locales=en_US'
+        ],
+      ),
+    ]);
+    final Directory arbDirectory = fileSystem.directory('arb')
+      ..createSync();
+    arbDirectory.childFile('foo.arb').createSync();
+    arbDirectory.childFile('bar.arb').createSync();
+
+    final LocalizationOptions options = LocalizationOptions(
+      header: 'HEADER',
+      headerFile: Uri.file('header'),
+      arbDirectory: Uri.file('arb'),
+      deferredLoading: true,
+      outputClass: 'Foo',
+      outputLocalizationsFile: Uri.file('bar'),
+      preferredSupportedLocales: 'en_US',
+      templateArbFile: Uri.file('example.arb'),
+      untranslatedMessagesFile: Uri.file('untranslated'),
+    );
+    final Depfile depfile = await generateLocalizations(
```

Comment:
If my understanding of `FakeProcessManager` is correct, maybe add a comment here stating that calling `generateLocalizations` should not fail since the passed in `LocalizationOptions` should be forwarded to match `FakeCommand` listed above.